atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1370,104 +1365,75 @@ private void commitWrites(TransactionService transactionService) {
-            // immutable timestamp hasn't moved forward - thorough sweep might sweep a conflict out from underneath us.
-            long microsForReadWriteConflictCheck = runAndReportTimeAndGetDurationMicros(
-                    () -> throwIfReadWriteConflictForSerializable(commitTimestamp),
-                    "readWriteConflictCheck");
-
-            // Verify that our locks and pre-commit conditions are still valid before we actually commit;
-            // this throwIfPreCommitRequirementsNotMet is required by the transaction protocol for correctness.
-            long microsForPreCommitLockCheck = runAndReportTimeAndGetDurationMicros(
-                    () -> throwIfImmutableTsOrCommitLocksExpired(commitLocksToken), "preCommitLockCheck");
-            long microsForUserPreCommitCondition = runAndReportTimeAndGetDurationMicros(
-                    () -> throwIfPreCommitConditionInvalid(commitTimestamp), "userPreCommitCondition");
-
-            long microsForPutCommitTs = runAndReportTimeAndGetDurationMicros(
-                    () -> putCommitTimestamp(commitTimestamp, commitLocksToken, transactionService),
-                    "commitPutCommitTs");
-
-            long microsSinceCreation = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis() - timeCreated);
-            getTimer("commitTotalTimeSinceTxCreation").update(microsSinceCreation, TimeUnit.MICROSECONDS);
-            getHistogram(AtlasDbMetricNames.SNAPSHOT_TRANSACTION_BYTES_WRITTEN).update(byteCount.get());
-            optionalProfile = Optional.of(ImmutableTransactionCommitProfile.builder()
-                    .startTimestamp(getTimestamp())
-                    .acquireRowLocksMicros(microsForRowLocks)
-                    .conflictCheckMicros(microsCheckingForConflicts)
-                    .writingToSweepQueueMicros(microsWritingToTargetedSweepQueue)
-                    .keyValueServiceWriteMicros(microsForWrites)
-                    .commitTimestampMicros(microsForGetCommitTs)
-                    .punchMicros(microsForPunch)
-                    .readWriteConflictCheckMicros(microsForReadWriteConflictCheck)
-                    .verifyPreCommitLockCheckMicros(microsForPreCommitLockCheck)
-                    .verifyUserPreCommitConditionMicros(microsForUserPreCommitCondition)
-                    .putCommitTimestampMicros(microsForPutCommitTs)
-                    .commitTimestamp(commitTimestamp)
-                    .totalCommitStageMicros(TimeUnit.NANOSECONDS.toMicros(commitStageTimer.stop()))
-                    .totalTimeSinceTransactionCreationMicros(microsSinceCreation)
-                    .build());
-        } finally {
-            long microsForPostCommitUnlock = runAndReportTimeAndGetDurationMicros(
-                    () -> timelockService.tryUnlock(ImmutableSet.of(commitLocksToken)), "postCommitUnlock");
+        timedAndTraced("commitStage", () -> {
```

Comment:
One thing this changes, if I'm reading it correctly, is that unsuccessful operations now go into the timer metrics because of try-with-resources on the timer context. This is probably reasonable - not tracking these previously was probably unintentional.

Reply:
yep, happy with